seatunnel-connectors-v2/connector-google-sheets/src/main/java/org/apache/seatunnel/connectors/seatunnel/google/sheets/source/SheetsSourceReader.java

```diff
@@ -0,0 +1,111 @@
+import java.io.IOException;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SheetsSourceReader extends AbstractSingleSplitReader<SeaTunnelRow> {
+
+    private SheetsParameters sheetsParameters;
+
+    private HttpRequestInitializer requestInitializer;
+
+    private static final String APPLICATION_NAME = "SeaTunnel Google Sheets";
+
+    private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+    private final SingleSplitReaderContext context;
+
+    private DeserializationSchema<SeaTunnelRow> deserializationSchema;
+
+    public SheetsSourceReader(SheetsParameters sheetsParameters, SingleSplitReaderContext context, DeserializationSchema<SeaTunnelRow> deserializationSchema) throws IOException {
+        this.sheetsParameters = sheetsParameters;
+        this.context = context;
+        this.deserializationSchema = deserializationSchema;
+    }
+
+    @Override
+    public void open() throws Exception {
+        byte[] keyBytes = Base64.getDecoder().decode(sheetsParameters.getServiceAccountKey());
+        ServiceAccountCredentials sourceCredentials = ServiceAccountCredentials
+            .fromStream(new ByteArrayInputStream(keyBytes));
+        sourceCredentials = (ServiceAccountCredentials) sourceCredentials
+            .createScoped(Collections.singletonList(SheetsScopes.SPREADSHEETS));
+        requestInitializer = new HttpCredentialsAdapter(sourceCredentials);
+
+    }
+
+    @Override
+    public void close() throws IOException {
+
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        Sheets service = new Sheets.Builder(httpTransport, JSON_FACTORY, requestInitializer)
+            .setApplicationName(APPLICATION_NAME)
+            .build();
+        ValueRange response = service.spreadsheets().values()
+            .get(sheetsParameters.getSheetId(), sheetsParameters.getSheetName() + "!" + sheetsParameters.getRange())
+            .execute();
+        List<List<Object>> values = response.getValues();
+        if (values != null) {
+            for (List<Object> row : values) {
+                Map<String, Object> data = new HashMap<>();
+                for (int i = 0; i < row.size(); i++) {
+                    String key = i + "";
+                    if (sheetsParameters != null && i < sheetsParameters.getHeaders().size()) {
+                        if (!sheetsParameters.getHeaders().get(i).equals("")) {
+                            key = sheetsParameters.getHeaders().get(i);
+                        }
+                    }
+                    data.put(key, row.get(i));
+                }
+                String dataStr = JsonUtils.toJsonString(data);
+                deserializationSchema.deserialize(dataStr.getBytes(), output);
```

Comment:
Thx. you should implement `GoogleSheetsDeserialization` yourself.

GoogleSheetsDeserialization using `seaTunnelRowType` to convert `List<Object> row` to SeaTunnelRow